includes/class-newspack-popups-settings.php

```diff
@@ -50,16 +50,134 @@ public static function create_admin_page() {
 	 * @param object $options options.
 	 */
 	public static function set_settings( $options ) {
-		if ( update_option( $options['option_name'], $options['option_value'] ) ) {
-			return self::get_settings();
+		$settings_list = self::get_settings();
+		$option_name   = $options['option_name'];
+		$setting       = array_reduce(
+			$settings_list,
+			function( $acc, $config ) use ( $option_name ) {
+				if ( $option_name === $config['key'] ) {
+					$acc = $config;
+				}
+				return $acc;
+			},
+			false
+		);
```

Comment:
I believe there's a duplicate effort here. Shouldn't the `get_setting_config()` method handle this? See the `update_setting()` method for reference on usage.

Reply:
It is a bit redundant, but the difference is that without the main Newspack plugin and its components, we don't have a `$section` to pass to these existing methods. I'm trying to avoid doing a big refactor for the sake of this "non-Newspack-UI"  settings page, which should hopefully be seen by only a tiny fraction of OSS users who choose to use Campaigns without the main Newspack plugin.

Reply:
Maybe rename the method to something like `set_setting_standalone()` and update its inline documentation so its purpose is more clear?

Reply:
That's a good idea. Updated in 06bd589. To further avoid confusion I also updated some method names on the Newspack Plugin side: https://github.com/Automattic/newspack-plugin/pull/1450#issuecomment-1033176235

Could use one last review of both commits just to make sure it all makes sense.